File: src/java/magma/locate/LastLocator.java
package magma.locate;

import java.util.Optional;

public record LastLocator(String infix) implements Locator {
    @Override
    public Optional<Integer> locate(String input) {
        final var index = input.lastIndexOf(infix());
        return index == -1 ? Optional.empty() : Optional.of(index);
    }

    @Override
    public String unwrap() {
        return this.infix;
    }

    @Override
    public int length() {
        return this.infix.length();
    }
}

File: src/c/magma/locate/LastLocator.c
#include <temp.h>
struct LastLocator {
	Rc_Optional<Integer> locate(void* _this_){
		struct LastLocator this = (struct LastLocator*) this;
		temp();
		temp();
	}
	Rc_String unwrap(void* _this_){
		struct LastLocator this = (struct LastLocator*) this;
		return value;
	}
	Rc_int length(void* _this_){
		struct LastLocator this = (struct LastLocator*) this;
		temp();
	}
};